proxy/src/main/java/com/velocitypowered/proxy/tablist/VelocityTabListLegacy.java

```diff
@@ -0,0 +1,93 @@
+package com.velocitypowered.proxy.tablist;
+
+import com.google.common.collect.ImmutableList;
+import com.velocitypowered.api.proxy.player.TabListEntry;
+import com.velocitypowered.api.util.GameProfile;
+import com.velocitypowered.proxy.connection.MinecraftConnection;
+import com.velocitypowered.proxy.protocol.packet.PlayerListItem;
+import com.velocitypowered.proxy.protocol.packet.PlayerListItem.Item;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import net.kyori.text.Component;
+
+public class VelocityTabListLegacy extends VelocityTabList {
+
+  private final Map<String, UUID> nameMapping = new ConcurrentHashMap<>();
+
+  public VelocityTabListLegacy(MinecraftConnection connection) {
+    super(connection);
+  }
+
+  @Override
+  public void setHeaderAndFooter(Component header, Component footer) {
+  }
+
+  @Override
+  public void clearHeaderAndFooter() {
+  }
+
+  @Override
+  public void addEntry(TabListEntry entry) {
+    super.addEntry(entry);
+    nameMapping.put(entry.getProfile().getName(), entry.getProfile().getId());
```

Comment:
Maybe the display name should be used instead, the display name is what is actually should be displayed on the client. This also supports (legacy) color formatting.

Reply:
The display name is a textcomponent and may be null. It is also transferred as a json, so I thought it would be better to fall back to non colorized and guarantee compatibility setting the displayname from the backend server.

Reply:
Then use the display name if present (converted to legacy formatting, with 16 char limit), and fall back to the profile name. That still guarantees the compatibility.